sdk/src/main/scala/com/horizen/block/NeighbourProof.scala

```diff
@@ -0,0 +1,59 @@
+package com.horizen.block
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonView}
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import com.horizen.serialization.{JsonMerklePathSerializer, Views}
+import com.horizen.utils.MerklePath
+import scorex.core.serialization.{BytesSerializable, ScorexSerializer}
+import scorex.util.serialization.{Reader, Writer}
+
+@JsonView(Array(classOf[Views.Default]))
+case class NeighbourProof
+(sidechainId: Array[Byte],
+ txsHash: Array[Byte],
+ wcertHash: Array[Byte],
+ leafIndex: Int,
```

Comment:
We can't believe in correctness of `leafIndex`. It should be calculated by verifier from scratch from `merklePath`.